### PR TITLE
[PyTorch] Add lr scheduler

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_optimization/lr-scheduler.md [hunk 1]
```diff
@@ -155,26 +246,11 @@ class SquareRootScheduler:
...
 ```{.python .input}
-scheduler = SquareRootScheduler(lr=1.0)
-d2l.plot(np.arange(num_epochs), [scheduler(t) for t in range(num_epochs)])
-```
-
-```{.python .input}
-#@tab tensorflow
-scheduler = SquareRootScheduler(1.0)
+#@tab all
+scheduler = SquareRootScheduler(lr=0.3)
```
Comment (AnirudhDagar) [inline chapter_optimization/lr-scheduler.md:253]: Updated the LR value to 0.3. This is required for stable training which can converge for PyTorch implementation when calling the train function below.

**Comment by mli** [comment]
Job d2l-en/PR-1305/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1305/

**Comment by mli** [comment]
Job d2l-en/PR-1305/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1305/

**Comment by mli** [comment]
Job d2l-en/PR-1305/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1305/

**Comment by mli** [comment]
Job d2l-en/PR-1305/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1305/

**Comment by mli** [comment]
Job d2l-en/PR-1305/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1305/

**Comment by mli** [comment]
Job d2l-en/PR-1305/8 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1305/

File: chapter_optimization/lr-scheduler.md [hunk 2]
```diff
@@ -356,15 +441,14 @@ In some cases initializing the parameters is not sufficient to guarantee a good
 A rather simple fix for this dilemma is to use a warmup period during which the learning rate *increases* to its initial maximum and to cool down the rate until the end of the optimization process. For simplicity one typically uses a linear increase for this purpose. This leads to a schedule of the form indicated below.
 
 ```{.python .input}
-scheduler = lr_scheduler.CosineScheduler(20, warmup_steps=5, base_lr=0.5,
+scheduler = lr_scheduler.CosineScheduler(20, warmup_steps=5, base_lr=0.3,
```
Comment (astonzhang) [inline chapter_optimization/lr-scheduler.md:444]: why do we change base_lr?

Comment (AnirudhDagar) [reply chapter_optimization/lr-scheduler.md:444]: PyTorch implementation was not converging regularly somehow when base_lr was 0.5. So I changed it across frameworks for consistency.

Comment (astonzhang) [reply chapter_optimization/lr-scheduler.md:444]: That's ok. Just give me a heads up if you change original mx code :)

File: chapter_optimization/lr-scheduler.md [hunk 3]
```diff
@@ -155,26 +244,11 @@ class SquareRootScheduler:
...
 ```{.python .input}
-scheduler = SquareRootScheduler(lr=1.0)
-d2l.plot(np.arange(num_epochs), [scheduler(t) for t in range(num_epochs)])
-```
-
-```{.python .input}
-#@tab tensorflow
-scheduler = SquareRootScheduler(1.0)
+#@tab all
+scheduler = SquareRootScheduler(lr=0.1)
```
Comment (astonzhang) [inline chapter_optimization/lr-scheduler.md:251]: Here's another example of original code modification.

File: chapter_optimization/lr-scheduler.md (hunk 2 above)
Comment (AnirudhDagar) [reply chapter_optimization/lr-scheduler.md:444]: Sure! I mentioned it [here](https://github.com/d2l-ai/d2l-en/pull/1305#discussion_r467208799). Will make it more explicit in the future if mxnet code is changed.

**Comment by AnirudhDagar** [comment]
Should we merge this now?
